Gathering and storing  of commits

TarasPavliuchyn opened the pull request:
added functionality of gathering and storing commits for each build, created pojo-models for commits and tickets, changed table "BUILDS" in db.

Commit:
added gathering and storing functionality of SCM metadata

Commit:
removed debug config

Commit:
done some refactoring

Commit:
done some refactoring

webdizz commented on plugin/src/main/java/com/epam/jenkins/deployment/sphere/plugin/metadata/model/SubMetaData.java:
```diff
@@ -0,0 +1,19 @@
+
+@Data
+@ExportedBean
+public class SubMetaData {
```
what does it mean Sub?

webdizz commented on plugin/src/main/java/com/epam/jenkins/deployment/sphere/plugin/utils/SubMetaDataExtractor.java:
```diff
@@ -0,0 +1,59 @@
+
+public class SubMetaDataExtractor {
+
+    public static String getMetaData(final AbstractBuild<?, ?> build) {
```
do not use statics, use IoC capabilities

webdizz replied:
method is with too complex logic, please refactor to several smaller methods with proper purpopse

webdizz commented on plugin/src/main/java/com/epam/jenkins/deployment/sphere/plugin/utils/SubMetaDataExtractor.java:
```diff
@@ -0,0 +1,59 @@
+            }
+            jsonMetaData = new ObjectMapper().writeValueAsString(metaData);
+        } catch (IOException e) {
+            e.printStackTrace();
```
do not use printStackTrace in production code

Commit:
Removed "Sub" part from  name of classes.
Removed static methods from utility class MetaDataExtractor. Used IoC capabilities.
Refactored complex logic of methods of  MetaDataExtractor.
Added logging of exception.

Commit:
added functionality of extracting MetaData from DB.

Commit:
done some refactoring

webdizz commented on plugin/src/main/java/com/epam/jenkins/deployment/sphere/plugin/metadata/persistence/query/BuildQuery.java:
```diff
@@ -73,8 +93,21 @@ public Build map(final int index, final ResultSet resultSet, final StatementCont
+            try {
+                metaData = new ObjectMapper().readValue(jsonString, MetaData.class);
+            } catch (IOException e) {
+                log.log(Level.SEVERE, "Failed deserializing Json to MetaData", e);
```
it's antipattern to log and throw, do one

webdizz commented on plugin/src/main/java/com/epam/jenkins/deployment/sphere/plugin/utils/MetaDataExtractor.java:
```diff
@@ -0,0 +1,61 @@
+    private MetaData collectMetaData(final AbstractBuild<?, ?> build) {
+        AbstractBuild<?, ?> tmpBuild = build;
+        MetaData metaData = new MetaData();
+        if (tmpBuild.getResult() == Result.SUCCESS) {
```
usually constant is from the left

webdizz commented on plugin/src/main/java/com/epam/jenkins/deployment/sphere/plugin/utils/MetaDataExtractor.java:
```diff
@@ -0,0 +1,61 @@
+
+    private List<Commit> getFailedBuildCommits(AbstractBuild<?, ?> tmpBuild) {
+        List<Commit> commits = new ArrayList<>();
+        while (tmpBuild != null && tmpBuild.getResult() != Result.SUCCESS) {
```
usually constant is from the left

webdizz commented on plugin/src/main/java/com/epam/jenkins/deployment/sphere/plugin/metadata/persistence/query/BuildQuery.java:
```diff
@@ -55,15 +63,27 @@ public void bind(final SQLStatement<?> q, final BindBuild bind, final Build arg)
+                try {
+                    jsonMetaData = new ObjectMapper().writeValueAsString(metaData);
+                } catch (IOException e) {
+                    log.log(Level.SEVERE, "Failed serializing MetaData to Json", e);
```
throw or log not both

Commit:
done integration of collection of metadata, fixed commented issues, fixed jira plugin error

Commit:
Done refactoring of Jira injection

webdizz commented:
please resubmit after fixing merge conflicts

Commit:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	plugin/src/main/java/com/epam/jenkins/deployment/sphere/plugin/metadata/collector/BuildVersionMetaDataCollector.java
	plugin/src/main/java/com/epam/jenkins/deployment/sphere/plugin/metadata/collector/InvolvedBuildChangesCollector.java
	plugin/src/main/java/com/epam/jenkins/deployment/sphere/plugin/metadata/jira/JiraMetaDataCollector.java
